ext/nnstreamer/extra/nnstreamer_grpc_flatbuf.cc

```diff
@@ -88,25 +89,154 @@ Status ServiceImplFlatbuf::_read_tensors (T reader)
 template <typename T>
 Status ServiceImplFlatbuf::_write_tensors (T writer)
 {
-  GstDataQueueItem *item;
-
-  /* until flushing */
-  while (gst_data_queue_pop (queue_, &item)) {
+  while (1) {
     Message<Tensors> tensors;
 
-    _get_tensors_from_buffer (GST_BUFFER (item->object), tensors);
-    writer->Write (tensors);
+    /* until flushing */
+    if (!fill_tensors (tensors))
+      break;
 
-    GDestroyNotify destroy = (item->destroy) ? item->destroy : g_free;
-    destroy (item);
+    writer->Write (tensors);
   }
 
   return Status::OK;
 }
 
+/** @brief convert tensors to buffer */
+void
+ServiceImplFlatbuf::_get_buffer_from_tensors (Message<Tensors> &msg,
+    GstBuffer **buffer)
+{
+  const Tensors *tensors = msg.GetRoot ();
+  guint num_tensor = tensors->num_tensor ();
+  GstMemory *memory;
+
+  *buffer = gst_buffer_new ();
+
+  for (guint i = 0; i < num_tensor; i++) {
+    const Tensor * tensor = tensors->tensor ()->Get (i);
+    const void * data = tensor->data ()->data ();
+    gsize size = VectorLength (tensor->data ());
+    gpointer new_data = g_memdup (data, size);
+
+    memory = gst_memory_new_wrapped (GST_MEMORY_FLAG_READONLY,
+        new_data, size, 0, size, new_data, g_free);
+    gst_buffer_append_memory (*buffer, memory);
+  }
+}
+
+/** @brief convert buffer to tensors */
+void
+ServiceImplFlatbuf::_get_tensors_from_buffer (GstBuffer *buffer,
+    Message<Tensors> &msg)
+{
+  MessageBuilder builder;
+
+  flatbuffers::Offset<flatbuffers::Vector<uint32_t>> tensor_dim;
+  flatbuffers::Offset<flatbuffers::String> tensor_name;
+  flatbuffers::Offset<flatbuffers::Vector<unsigned char>> tensor_data;
+  flatbuffers::Offset<Tensor> tensor;
+  flatbuffers::Offset<Tensors> tensors;
+  std::vector<flatbuffers::Offset<Tensor>> tensor_vector;
+  Tensor_type tensor_type;
+
+  unsigned int num_tensors = config_->info.num_tensors;
+  frame_rate fr = frame_rate (config_->rate_n, config_->rate_d);
+
+  GstMapInfo map;
+  gsize data_ptr = 0;
+
+  gst_buffer_map (buffer, &map, GST_MAP_READ);
+
+  for (guint i = 0; i < num_tensors; i++) {
+    const GstTensorInfo * info = &config_->info.info[i];
+    gsize tsize = gst_tensor_info_get_size (info);
+
+    tensor_dim = builder.CreateVector (info->dimension, NNS_TENSOR_RANK_LIMIT);
+    tensor_name = builder.CreateString ("Anonymous");
+    tensor_type = (Tensor_type) info->type;
+    tensor_data = builder.CreateVector<unsigned char> (map.data + data_ptr, tsize);
```

Comment:
line 160 : condition to check data size after creating vector

Reply:
Fixed, thanks.